pkg/cmd/cli/cmd/newapp.go

```diff
@@ -483,6 +483,9 @@ func setupAppConfig(f *clientcmd.Factory, out io.Writer, c *cobra.Command, args
 	if len(config.SourceImage) == 0 && len(config.SourceImagePath) != 0 {
 		return kcmdutil.UsageError(c, "--source-image must be specified when --source-image-path is specified.")
 	}
+	if len(config.Strategy) != 0 && config.Strategy != "docker" && config.Strategy != "source" {
```

Comment:
in addition to constantizing these, should this be moved to the place where (I assume) we're special casing the "docker" strategy, and instead make it a switch of the strategies we handle, with a default that returns an error? If we add support for another strategy, it would be nice to just update in one place.